rust/lance-index/Cargo.toml

```diff
@@ -69,6 +69,8 @@ async-channel = "2.3.1"
 bitpacking = { version = "0.9.2", features = ["bitpacker4x"] }
 rand_distr.workspace = true
 lance-datagen.workspace = true
+rangemap.workspace = true
+dashmap = "6"
```

Comment:
Is there any particular reason we need dashmap here?  Just wondering if we can get away without adding a new dependency.